utils/subset.py

```diff
@@ -0,0 +1,128 @@
+import argparse
+import logging
+
+import hail as hl
+
+from gnomad.utils.filtering import subset_samples_and_variants
+
+
+logging.basicConfig(
+    format="%(asctime)s (%(name)s %(lineno)s): %(message)s",
+    datefmt="%m/%d/%Y %I:%M:%S %p",
+)
+logger = logging.getLogger("subset")
+logger.setLevel(logging.INFO)
+
+
+def remap_sample_ids(mt: hl.MatrixTable, remap_path: str):
+    """
+    Maps the MatrixTable's sample ID field ('s') to the sample ID used within seqr ('seqr_id').
+
+    If the sample does not have a mapping in the remap file, their ID becomes their seqr ID.
+
+    :param hl.MatrixTable mt: Input MatrixTable.
+    :param str remap_path: Path to a file with two columnsL 's' and 'seqr_id'.
+    :return: MatrixTable with VCF sample IDs mapped to seqr IDs and keyed with seqr ID.
+    :rtype: hl.MatrixTable
+    """
+    remap_ht = hl.import_table(remap_path, key="s")
+    missing_samples = remap_ht.anti_join(mt.cols()).collect()
+    remap_count = remap_ht.count()
+
+    if len(missing_samples) != 0:
+        logger.error(
+            f"Only {remap_ht.semi_join(mt.cols()).count()} out of {remap_count} "
+            "remap IDs matched IDs in the variant callset.\n"
+            f"IDs that aren't in the callset: {missing_samples}\n"
+            f"All callset sample IDs:{mt.s.collect()}",
+            missing_samples,
+        )
+
+    mt = mt.annotate_cols(**remap_ht[mt.s])
+    remap_expr = hl.cond(hl.is_missing(mt.seqr_id), mt.s, mt.seqr_id)
+    mt = mt.annotate_cols(seqr_id=remap_expr, vcf_id=mt.s)
+    mt = mt.key_cols_by(s=mt.seqr_id)
+    logger.info(f"Remapped {remap_count} sample ids...")
+    return mt
+
+
+def main(args):
+    """
+    Subset joint-called VCF or MT to desired samples.
+
+    Used when CMG/MGRC collaborators request VCFs of their data.
+    """
+    hl.init(log="/subset.log", default_reference="GRCh38")
+
+    if args.vcf_path:
+        logger.info("Importing VCF...")
+        logger.warning("Assuming VCF is bgzipped!")
+        mt = hl.import_vcf(
+            args.vcf_path, force_bgz=True, reference_genome=args.import_build
+        )
+
+    else:
+        logger.info("Reading in MT...")
+        mt = hl.read_matrix_table(args.mt_path)
+
+    logger.info(f"Input MT counts: {mt.count()}")
+    mt.describe()
+
+    if args.mapping:
+        logger.info("Mapping VCF IDs to seqr IDs...")
+        logger.warning("Assuming mapping file has header!")
```

Comment:
How about
```suggestion
        logger.warning("Assuming mapping file the field names s and seqr_id!")
```
or something indicating the field names needed